### PR TITLE
Dynamically register/unregister client capability

### LINKED ISSUE #58: Dynamically register/unregister client capability
Add API (_sync_ and _async_) to _protocol_ and _server_ for dynamic (un)registration of client capabilities.

### LINKED ISSUE #68: Update language server protocol to v3.14.0
Update [features.py](https://github.com/openlawlibrary/pygls/blob/master/pygls/features.py) and [types.py](https://github.com/openlawlibrary/pygls/blob/master/pygls/types.py) to support the [current LSP specification](https://microsoft.github.io/language-server-protocol/specification#version_3_14_0).

### PR DESCRIPTION
**Pull request by danixeee**
## Description (e.g. "Related to ...", etc.)

Closes #58.
Closes #68.

This PR also closes #68 because I had to fix some field names from protocol specification when working on (#58).

## Code review checklist (for code reviewer to complete)

- [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [ ] Title summarizes what is changing
- [ ] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated, as appropriate
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Standalone docs have been updated accordingly
- [ ] [CONTRIBUTORS.md][contributors] was updated, as appropriate
- [ ] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/pygls/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/
[contributors]: https://github.com/openlawlibrary/pygls/blob/master/CONTRIBUTORS.md

### CODE REVIEW

File: pygls/types.py [hunk 1]
```diff
@@ -1270,8 +1282,8 @@ def __init__(self, id: str, method: str):
 
 
 class UnregistrationParams:
-    def __init__(self, unregistrations: List[Unregistration]):
-        self.unregistration = unregistrations
+    def __init__(self, unregisterations: List[Unregistration]):
```
Comment (renatav) [inline pygls/types.py]: Renamed unregistrations to unregisterations. Was this accidental?

File: pygls/types.py [hunk 2]
```diff
@@ -687,6 +687,18 @@ def __init__(self, uri: str, range: 'Range'):
         self.range = range
 
 
+class LocationLink:
```
Comment (renatav) [inline pygls/types.py]: Should this and other changes made in order to support the current LSP specification be mentioned in the changelog?

File: pygls/protocol.py [hunk 3]
```diff
@@ -593,7 +612,7 @@ def bf_workspace__execute_command(self,
         cmd_handler = self.fm.commands[params.command]
         self._execute_request(msg_id, cmd_handler, params.arguments)
 
-    def get_configuration(self, params, callback=None):
+    def get_configuration(self, params, callback):
```
[the remark this answers is not in the document]
Comment (danixeee) [reply pygls/protocol.py]: `get_configuration` is supposed to return a value. That value is passed as an argument to the callback function when result is returned from the client. In other words, there is no reason to call this function without doing something with returned value.

File: pygls/types.py (hunk 1 above)
Comment (danixeee) [reply pygls/types.py]: It was really hard to find why it wasn't working and the reason was that [specification](https://microsoft.github.io/language-server-protocol/specification#client_unregisterCapability) says "unregisterations" (!?) :)

**Comment by danixeee** [comment]
I have created another [issue](https://github.com/openlawlibrary/pygls/issues/68) which will be closed with this PR as well and updated the CHANGELOG.
I won't merge this PR until I test it on _Codify_.

File: pygls/types.py (hunk 1 above)
Comment (renatav) [reply pygls/types.py]: That must've been frustrating.

**Pull request merged**